task/backend/analytical_storage.go

```diff
@@ -29,6 +30,71 @@ const (
 	statusTag = "status"
 )
 
+func CreateDestination(log *zap.Logger, cli influxdb.Client, dest DataDestination) error {
+	const numTries = 5
+	for try := 0; try < numTries; try++ {
+		// poll for influxdb to come up - usually it should be already up by the time we get here.
+		if try > 0 {
+			time.Sleep(1 * time.Second)
+		}
+		queryForError := func(q string) error {
+			// This query is only successful if the bucket exists
+			result, err := cli.QueryFlux(influxdb.FluxQuery{
+				Org:   dest.Org,
+				OrgID: dest.OrgID,
+				Query: q,
+			})
+			if err != nil {
+				return err
+			}
+			for result.More() {
+				if err := result.Next().Tables().Do(func(table flux.Table) error {
+					return table.Do(func(col flux.ColReader) error {
+						return nil
+					})
+				}); err != nil {
+					return err
+				}
```

Comment:
If I'm reading this loop correctly, if we only succeed in creating the bucket on the last attempt, we will actually return an error even if the bucket was created. The bucket check and no error return is done at the top of the loop, and on the last iteration we can't get back to the top to check for success. Doesn't seem likely to happen in the real life, but might be confusing when it does.

Reply:
fixed